Interop/Runtime/src/main/kotlin/kotlinx/cinterop/Types.kt

```diff
@@ -67,26 +67,32 @@ abstract class CPointed(rawPtr: NativePtr) : NativePointed(rawPtr.toNonNull())
  * There are also other implementations of [CValuesRef] that provide temporary pointer,
  * e.g. Kotlin Native specific [refTo] functions to pass primitive arrays directly to native.
  */
-abstract class CValuesRef<T : CPointed> {
+public abstract class CValuesRef<T : CPointed> {
     /**
-     * If this reference is [CPointer], returns this pointer.
-     * Otherwise copies the referenced values to [placement] and returns the pointer to the copy.
+     * If this reference is [CPointer], returns this pointer, otherwise
+     * allocate storage value in the scope and return it.
      */
-    abstract fun getPointer(scope: AutofreeScope): CPointer<T>
+    public abstract fun getPointer(scope: AutofreeScope): CPointer<T>
+
+    /**
+     * Copy the referenced values to [placement] and returns the pointer to it for values, or just return [this]
+     * pointer if this value is already a CPointer.
+     */
+    public abstract fun place(placement: CPointer<T>): CPointer<T>
```

Comment:
This method should be declared in `CValues` instead since it makes no sense for other `CValuesRef` kinds.
Also `CValues` can have default `getPointer` implementation using `place` and `size`.
Its subclasses may still require their own `getPointer` implementations to optimize out the virtual calls.